### ISSUE TITLE
'WAIT' Issue

### ISSUE THREAD

**Issue by illudota** [GC-spigot/AdvancedEnchantments#2715]
### Describe the bug

Wait effect still procs even though player has already died, this can result to CEs with 'WAIT' effect still proccing even tho player has already died.

### How to reproduce

Create a 'WAIT' CE and add messages or smth, then use it on yourself then kill yourself instantly, well u will get it once i show u the vid below, really dangerous since some Bleed CEs still proc even after death, so example if a player dies exactly the bleed CE procced, he'll die again in spawn due to the 'WAIT' and 'HARM' effects still ongoing, sometimes when players with items with holy white scroll die, they'll die again in spawn, resulting to dropping their precious item in spawn, just like the vid below.

### Screenshots / Videos

https://www.youtube.com/watch?v=_zdrT_fmC34

CE used in video(Which i use in my server):
```rupture:
display: '%group-color%Rupture'
description: "&fBleeds, weakens removes regen\n&ffrom your opponent\n&fafter the bleed duration, there will be\n&fa slight burst of damage\n&6Requires:\n&eDeep &eBleed &eVI"
applies-to: 'Swords'
group: 'MASTERY'
type: 'ATTACK'
applies:
- ALL_SWORD
settings:
required-enchants:
- 'deepbleed:6'
removed-enchants:
- 'deepbleed'
levels:
'1':
cooldown: 10
effects:
- 'CURE:REGENERATION %victim%'
- 'POTION:WEAKNESS:0:100 %victim%'
- 'MESSAGE:&4&l** RUPTURE ** %attacker%'
- 'MESSAGE:&4&l** RUPTURED ** %victim%'
- 'PLAY_SOUND_OUTLOUD:ENTITY_STRIDER_DEATH'
- 'ADD_HARM:1-4</random number> %victim%'
- 'WAIT:10'
- 'ADD_HARM:1-4</random number> %victim%'
- 'WAIT:10'
- 'ADD_HARM:1-4</random number> %victim%'
- 'WAIT:10'
- 'ADD_HARM:1-4</random number> %victim%'
- 'WAIT:10'
- 'ADD_HARM:1-4</random number> %victim%'
- 'WAIT:10'
- 'ADD_HARM:1-4</random number> %victim%'
- 'WAIT:10'
- 'ADD_HARM:1-4</random number> %victim%'
- 'WAIT:10'
- 'ADD_HARM:1-4</random number> %victim%'
- 'WAIT:10'
- 'ADD_HARM:1-4</random number> %victim%'
- 'WAIT:10'
- 'ADD_HARM:1-4</random number> %victim%'
- 'WAIT:10'
- 'ADD_HARM:1-4</random number> %victim%'
- 'WAIT:10'
- 'ADD_HARM:10 %victim%'
```

### "/ae plinfo" link

https://paste.md-5.net/racezuxovi

### Server Log

https://mcpaste.io/d756f95675b584fa

**Comment by Burchard36** [comment]
Hello, I looked into this issue and was able to reproduce this effect on my end.

I will look into a list of possible fix for this issue and update you accordingly, thank you for the bug report!

**Comment by Burchard36** [comment]
Hello there, fix as been implemented and will be available hopefully next bug patch update!

Video: https://www.youtube.com/watch?v=Ij5gF_m9KdE

**Comment by Burchard36** [comment]
Fix will be implemeneted when 8.9.9, have a good day!
